### PR TITLE
Remove networkId

### PR DESCRIPTION
**Pull request by jiexi**
Wallets shouldn't be directly concerned about the network ID as this more of a p2p concept for gossip. What wallets really care about is chain ID as that is the correct value to use to identify a chain, build transactions, etc. Although these two values usually match (ignoring hex/dec formatting), there are [exceptions](https://medium.com/@pedrouid/chainid-vs-networkid-how-do-they-differ-on-ethereum-eec2ed41635b).

We want to remove usage of `networkId` from the SmartTransactionController.

* Fixes [mmp-1068](https://github.com/MetaMask/MetaMask-planning/issues/1068)
* See: [core-1633](https://github.com/MetaMask/core/pull/1633)
* See: https://github.com/MetaMask/metamask-extension/pull/20652

### CODE REVIEW

File: package.json
Comment (jiexi) [inline package.json:1]: reminder to update this after core changes merged

File: src/SmartTransactionsController.ts
Comment (mcmire) [inline src/SmartTransactionsController.ts:578]: Are we sure that in all places where `metamaskNetworkId` is referred to in the extension, chain ID is used instead?

Comment (jiexi) [reply src/SmartTransactionsController.ts:578]: Uncertain how easy it will be to fully transition away from `metamaskNetworkId` for old transactions that predate `chainId`.

See https://github.com/MetaMask/metamask-extension/pull/20652#issuecomment-1707351348

Comment (jiexi) [reply src/SmartTransactionsController.ts:578]: This is now resolved on extension side

**Review by dan437 — APPROVED** [review]
That worked!

**Review by shanejonas — APPROVED** [review]

**Comment by BelfordZ** [comment]
Needs the previews replaced with the releases, but looks good otherwise

**Comment by jiexi** [comment]
Blocked until https://github.com/MetaMask/core/pull/1633 is merged and released

File: package.json
Comment (jiexi) [reply package.json:29]: https://github.com/MetaMask/core/pull/1633

Bumping NetworkController to v12.2.0 should be fine

File: CHANGELOG.md
Comment (adonesky1) [inline CHANGELOG.md:9]: not that it matters since this will be a major dep bump anyways but why is this breaking in the context of this package?

**Review by adonesky1 — DISMISSED** [review]
One non-blocking question

File: CHANGELOG.md
Comment (jiexi) [reply CHANGELOG.md:9]: Great catch. I guess technically it isn't breaking because it doesn't directly use the breaking changes from NetworkController. Only thing referenced is the `NetworkState` type, and it doesn't touch `networkId` there. I'll remove the BREAKING

Comment (jiexi) [reply CHANGELOG.md]: Done: https://github.com/MetaMask/smart-transactions-controller/pull/191/commits/85b5b7efbdf8c1431f72ff8f8554f71e7c990a2f

**Review by adonesky1 — APPROVED** [review]

**Pull request merged**
